malloc_trim gave error when building Android executable files.

Hello, I was trying to build vvenc to target Android (aarch64 build) cross compile from Ubuntu 22.04 LTS.

Version: Android NDK 24

Used build commands:
```
cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_SYSTEM_NAME=Android -DCMAKE_ANDROID_ARCH_ABI=arm64-v8a -DCMAKE_TOOLCHAIN_FILE=${HOME}/android-ndk-r24/build/cmake/android.toolchain.cmake -DANDROID_STL=c++_static -DANDROID_PLATFORM=android-$MINSDKVERSION .
```

The error gave me when build is almost done:
```
/home/martineesmaa/vvenc/source/Lib/vvenc/vvencimpl.cpp:229:3: error: use of undeclared identifier 'malloc_trim'

  malloc_trim(0);   // free unused heap memory

  ^
/home/martineesmaa/vvenc/source/Lib/vvenc/vvencimpl.cpp:427:3: error: use of undeclared identifier 'malloc_trim'

  malloc_trim(0);   // free unused heap memory

  ^
2 errors generated.
gmake[2]: *** [source/Lib/vvenc/CMakeFiles/vvenc.dir/build.make:1084: source/Lib/vvenc/CMakeFiles/vvenc.dir/vvencimpl.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:198: source/Lib/vvenc/CMakeFiles/vvenc.dir/all] Error 2
gmake: *** [Makefile:146: all] Error 2
```

When I removed code following: malloc_trim. It worked well after malloc_trim is removed in only C++ file.

- Martin Eesmaa

--- a/source/Lib/vvenc/vvencimpl.cpp
+++ b/source/Lib/vvenc/vvencimpl.cpp
@@ -229,7 +229,7 @@ int VVEncImpl::uninit()
 #endif
   }
 
-#if defined( __linux__ )
+#if defined( __linux__ ) && defined( __GLIBC__ )
   malloc_trim(0);   // free unused heap memory
 #endif
 
@@ -427,7 +427,7 @@ int VVEncImpl::encode( vvencYUVBuffer* pcYUVBuffer, vvencAccessUnit* pcAccessUni
     iRet = xCopyAu( *pcAccessUnit, cAu  );
   }
 
-#if defined( __linux__ )
+#if defined( __linux__ ) && defined( __GLIBC__ )
   malloc_trim(0);   // free unused heap memory
 #endif
 

